libraries/pi4j-library-linuxfs/src/main/java/com/pi4j/library/linuxfs/LinuxFile.java

```diff
@@ -214,7 +214,8 @@ protected synchronized IntBuffer getOffsetsBuffer(int size) {
 
         if (byteSize > localBufferSize)
             throw new ScratchBufferOverrun();
-
+        // if no buffer currently exists, new one always created.
+        // if buffer exists and limit is not equal to this getOffsetsBuffer, allocate new
         if (buf == null) {
             ByteBuffer bb = ByteBuffer.allocateDirect(localBufferSize);
 
```

Comment:
Robert.  U asked about reducing this code. The nativeOrder does alter the buffer and should remain.   I did not challenge the comment that nativeOrder be set prior to making the buffer an IntBuff
```suggestion

```